### PR TITLE
Add release notes for 0.248

### PR DESCRIPTION
**Pull request by prithvip**
# Missing Release Notes
## Stephen Dimmick
- [x] https://github.com/prestodb/presto/pull/15666 Add scalar function DESTRUCTURE_TDIGEST (Merged by: Timothy Meehan)

## Tim Meehan
- [x] https://github.com/prestodb/presto/pull/15679 Revert "Add cpu and memory stats to NodeTaskMap" (Merged by: Timothy Meehan)

## Vic Zhang
- [x] https://github.com/prestodb/presto/pull/15629 Optimize empty bucket file creation for temporary table (Merged by: Andrii Rosa)
- [x] https://github.com/prestodb/presto/pull/15629 Optimize empty bucket file creation for temporary table (Merged by: Andrii Rosa)
- [x] https://github.com/prestodb/presto/pull/15629 Optimize empty bucket file creation for temporary table (Merged by: Andrii Rosa)

# Extracted Release Notes
- #15588 (Author: Sreeni Viswanadha): Implement MAP_UNION_SUM aggregation
  - :func `map_union_sum`.
- #15626 (Author: James Petty): Reduce output buffer memory manager contention
  - Improve output buffer memory tracking to reduce lock contention.
- #15636 (Author: Chi Tsai): Add warning when OR is found in JOIN
  - In StatementAnalyzer.java, checked if the JOIN expression needed a warning and created one if necessary.
  - In TestAnalyzer, added JUnit tests that check for warning when OR is directly in predicate and no warning when OR not present.
- #15654 (Author: Chi Tsai): Add coordinator REST endpoint for TaskInfo
  - Added a REST endpoint to get TaskInfo without directly going to worker endpoint.
- #15655 (Author: Luca): Allow to set a truststore password in InternalCommunicationConfig
  - New configuration parameter `internal-communication.https.trust-store-password` to set the Java Truststore password used for https in internal communications between nodes.
- #15672 (Author: Arunachalam Thirupathi): Configurable ZSTD compression level for ORC Writer
  - Configurable ZSTD compression Level.
- #15687 (Author: Leiqing Cai): Allow multiple overloads of an UDF to be dropped at the same time
  - Add support to drop multiple UDFs at the same time.
- #15702 (Author: Lung-Yen Chen): Export output table names in QueryInfo
  - Add output table names to Presto Verifier's outputs.
- #15716 (Author: Ariel Weisberg): Allow session property rules to override
  - Add support for overriding session properties using session property managers :doc:`/admin/session-property-managers`. Setting `overrideSessionProperties` to true will cause the property to be overridden and remain overridden even if subsequent rules match the property but don't have `overrideSessionProperties` set.

# All Commits
- c3454a9445b077cbff1f95d3269f791ea5e9c912 Support exception stack and error code in remote function execution (Leiqing Cai)
- cb63da2b98f72c0edbb9724c2692b5cb0956c533 Add a warning for JOIN with an OR in the predicate (Chi Tsai)
- 49b901ee38529e99576895c825f53783a163c3f4 Allow session property rules to override (Ariel Weisberg)
- 04626d6a9abdf87abbf6e3ab67eb8b13fb56f7d8 Add scalar function DESTRUCTURE_TDIGEST (Stephen Dimmick)
- 646648e5b4bbdd24ba6a8fac78032538957c4e68 Allow multiple overloads of an UDF to be dropped at the same time (Leiqing Cai)
- c71cab1b16f8f7cfae683eb26e55aadc1c71640a Generate shorter aggregation function names (Rongrong Zhong)
- c344d7bca4aad331c250efd1a4ed40f703d644b3 Limit the codegen class name length (Rongrong Zhong)
- f6bd37f079de46073d0f38fe58580183e79c8222 Add table path to HiveTableLayoutHandle (Vic Zhang)
- 45828a194b7970b7bf8bb927f128c92d74794d4d Add path and isCreateTable to StagingFileCommitter (Vic Zhang)
- f7557ad1dbfcdfd90319f1c96241c434e7c4f5fc Mark new partitions for new partitioned table (Vic Zhang)
- 09c5b096061bb22f7b086639db7f30033957395c Refactor declareIntentionToWrite to include HdfsContext (Vic Zhang)
- eed7b37e7aba00eca745fcf62e837adf6822504b Refactor dropTable in SemiTransactionHiveMetastore (Vic Zhang)
- af8f69e99f03a53be3d66daf38c53b009b9da657 Refactor HdfsContext used by HiveLocationService (Vic Zhang)
- 029f8096e67267e70cfc3458f3fd96f8ced48e9e Add isNewTable and tablePath to HdfsContext (Vic Zhang)
- 77e7d3cd3975f50fe52d86afecb503815ac72fc8 Load token_service by spark class loader (Vic Zhang)
- cbb19ffcab47156a08e92059d604b25f18aea26c Add Drift annotations for BasicQueryInfo and NodeStatus serialization (Tim Meehan)
- ac79506204bcab012acdaba4263585ed28502a80 Allow multiple coordinators to update memory assignments (Tim Meehan)
- af32e7f0a6872e7b714bf89bbcc35c52af74a6e4 Allow multiple coordinators in DistributedQueryRunner (Tim Meehan)
- 05b6200c376d50462f737cbb4f7dc71659426692 Add ResourceManager (Tim Meehan)
- ae32d32cb34c17cf6f3b2846d5bf65c7f07f5998 Allow node to be identified as Resource Manager (Tim Meehan)
- 863895858551e6198eb839fb7f36e78e1210009a Include exception for HIVE_FILE_NOT_FOUND error (Vic Zhang)
- a6dc72d33518e6697914a4798f0a30037216dc7f Add session property to create empty bucket files (Vic Zhang)
- 95daf0f38ba301a4f010bcefdf110ec77879b92b Optimize empty bucket file creation for temporary table (Vic Zhang)
- ab60a389a8d66a8eabb0fa5626f6511aa13dcb43 Export output table names in QueryInfo (Lung-Yen Chen)
- 23f464606544cb01ecd845deb9038ed65880c55a Add a dummy class to presto-thrift-spec (Leiqing Cai)
- daa287d32e46092256d8771d28d11881a474a6ac Remove test "overrides" that are no longer relevant (Rebecca Schlussel)
- bbda25ba8b946c1292418ce06f5cd8bdf65c4d81 Disable testCorrelatedNonAggregationScalarSubqueries with spill (Rebecca Schlussel)
- 69ac888b7549d0c920806aff0ce21f9e4eda8969 Use long for partitionVersion (Nikhil Collooru)
- fa1a9597f18a4089c301ce63cd8fd6bf7b53a3ba Implement MAP_UNION_SUM aggregation (Sreeni Viswanadha)
- 4472e9cc6996b174f21bb8c57f339d0c4a731256 Add coordinator REST endpoint for TaskInfo (Chi Tsai)
- 733ab430e21579c5b370b9cc0f2ccb3e0e8528e2 Add Kafka Json encoder (Yang Yang)
- 04b8abfceab34e9a8848e6071481192cfbed80d0 Revert "Add cpu and memory stats to NodeTaskMap" (Timothy Meehan)
- 80b9fd5df8537b3960b137bace7158f101247f5f Configurable ZSTD compression level for ORC Writer (Arunachalam Thirupathi)
- 00b465cff52848fdabb2d1034ed2ce9dc6672ab4 Evict metastore caching when partition version does not match (Nikhil Collooru)
- d401553d037e94df556f0b1b352c4966ffc1e544 Refactor the OrcWriter Stats (Arunachalam Thirupathi)
- 641347b4ded84e4ae1587dbc767ec5de29da9d7b Reduce lock contention in OutputBufferMemoryManager (James Petty)
- dff92fe4d56071dcf7230764ea10f911541f1611 Allow to set a truststore password in InternalCommunicationConfig (Luca)
- c48825e08a5cffaf100828975f1d5255d87823c7 Fix maven scope for airlift units library (Ajay George)
- f662badb5ed2c21f74b12a95de436993e00e6299 First-draft of Presto client protocol documentation (David Stryker)
- 859f594241e189469ca92c6f01641ab3d12e6f3e Handle UnknownType in TypeVariableConstraint (Sreeni Viswanadha)
- 4bd2aded74aa97055d4a51053ff1cb4fd05cc1da Add extra information in CacheQuota for Alluxio to use (Ke Wang)

### CODE REVIEW

**Review by mayankgarg1990 — CHANGES_REQUESTED** [review]
please review https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines and accordingly incorporate everything mentioned there in your release notes PR

File: presto-docs/src/main/sphinx/release/release-0.248.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+* Add a REST endpoint to get TaskInfo, without directly going to worker endpoint.
+* Add new aggregation function :func:`map_union_sum`.
+* Add new scalar function DESTRUCTURE_TDIGEST to expose the internal TDigest data structure as a ROW
+* Add support to configure ZSTD compression level for ORC writer.
+* Add warning for JOIN conditions with OR expressions.
+* Add configuration property `internal-communication.https.trust-store-password` to set the Java Truststore password used for https in internal communications between nodes.
+
+Hive Connector Changes
+________________
+* Add session property `temporary_table_create_empty_bucket_files` and configuration property `hive.create-empty-bucket-files-for-temporary-table`, which, when set to false, disables the creation of zero-row files for temporary table empty buckets, to improve performance.
```
Comment (rschlussel) [inline presto-docs/src/main/sphinx/release/release-0.248.rst:28]: the keywords should be in double angle quotes

File: presto-docs/src/main/sphinx/release/release-0.248.rst [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+* New scalar function DESTRUCTURE_TDIGEST.
+* Support for session property managers, that can override session properties set by a user.
+
+**Details**
+===========
+
+General Changes
+_______________
+* Improve output buffer memory tracking to reduce lock contention.
+* Add support for overriding session properties using session property managers :doc:`/admin/session-property-managers`. Setting `overrideSessionProperties` to true will cause the property to be overridden and remain overridden even if subsequent rules match the property but don't have `overrideSessionProperties` set.
```
Comment (rschlussel) [inline presto-docs/src/main/sphinx/release/release-0.248.rst:17]: overrideSessionProperties should be in double angle quotes.

File: presto-docs/src/main/sphinx/release/release-0.248.rst [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+* Add support to drop multiple UDFs at the same time.
+* Add a REST endpoint to get TaskInfo, without directly going to worker endpoint.
```
Comment (rschlussel) [inline presto-docs/src/main/sphinx/release/release-0.248.rst:19]: Is this documented anywhere?  what's the endpoint?

File: presto-docs/src/main/sphinx/release/release-0.248.rst [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+* Add new aggregation function :func:`map_union_sum`.
+* Add new scalar function DESTRUCTURE_TDIGEST to expose the internal TDigest data structure as a ROW
```
Comment (rschlussel) [inline presto-docs/src/main/sphinx/release/release-0.248.rst:21]: Is this documented anywhere?  if so, it should have a link to the function documentation like the other function release notes.  If not, we should create a task for whoever added it to add documentation.

File: presto-docs/src/main/sphinx/release/release-0.248.rst (hunk 1 above)
Comment (prithvip) [reply presto-docs/src/main/sphinx/release/release-0.248.rst]: done

File: presto-docs/src/main/sphinx/release/release-0.248.rst (hunk 2 above)
Comment (prithvip) [reply presto-docs/src/main/sphinx/release/release-0.248.rst]: done

File: presto-docs/src/main/sphinx/release/release-0.248.rst (hunk 3 above)
Comment (prithvip) [reply presto-docs/src/main/sphinx/release/release-0.248.rst]: There's no documentation but I've added the endpoint to the release notes

File: presto-docs/src/main/sphinx/release/release-0.248.rst (hunk 4 above)
Comment (prithvip) [reply presto-docs/src/main/sphinx/release/release-0.248.rst]: There's no documentation for any TDigest function at all, not just this one

Comment (prithvip) [reply presto-docs/src/main/sphinx/release/release-0.248.rst]: I found this issue https://github.com/prestodb/presto/issues/15674, so looks like it's being tracked

File: presto-docs/src/main/sphinx/release/release-0.248.rst [hunk 5]
```diff
@@ -0,0 +1,37 @@
+=============
+Release 0.248
+=============
+
+**Highlights**
+==============
+* New aggregation function MAP_UNION_SUM.
```
Comment (mayankgarg1990) [inline presto-docs/src/main/sphinx/release/release-0.248.rst:7]: https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines#formatting

```
:func:`map_union_sum`
```

File: presto-docs/src/main/sphinx/release/release-0.248.rst [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+* New scalar function DESTRUCTURE_TDIGEST.
+* Support for session property managers, that can override session properties set by a user.
```
Comment (mayankgarg1990) [inline presto-docs/src/main/sphinx/release/release-0.248.rst:9]:
```
Add support for overriding session properties using session property managers. See :doc:`/admin/session-property-managers`

File: presto-docs/src/main/sphinx/release/release-0.248.rst [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+**Details**
+===========
+
+General Changes
+_______________
+* Improve output buffer memory tracking to reduce lock contention.
```
Comment (mayankgarg1990) [inline presto-docs/src/main/sphinx/release/release-0.248.rst:16]: Start with `Optimize` and also - how does this help ? Does this improve all queries or some specific set of queries?

File: presto-docs/src/main/sphinx/release/release-0.248.rst [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+* Add support for overriding session properties using session property managers :doc:`/admin/session-property-managers`. Setting ``overrideSessionProperties`` to true will cause the property to be overridden and remain overridden even if subsequent rules match the property but don't have ``overrideSessionProperties`` set.
+* Add support to drop multiple UDFs at the same time.
+* Add a REST endpoint ``/v1/taskInfo/{{taskId}}`` on the coordinator to get TaskInfo without needing to go directly to the worker's endpoint.
+* Add new aggregation function :func:`map_union_sum`.
+* Add new scalar function DESTRUCTURE_TDIGEST to expose the internal TDigest data structure as a ROW.
```
Comment (mayankgarg1990) [inline presto-docs/src/main/sphinx/release/release-0.248.rst:21]:
```
Add new scalar function :func:`destructure_tdigest` to expose the internal ``TDigest`` data structure as a ``ROW``
```

File: presto-docs/src/main/sphinx/release/release-0.248.rst [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+* Add support to configure ZSTD compression level for ORC writer.
+* Add warning for JOIN conditions with OR expressions.
+* Add configuration property ``internal-communication.https.trust-store-password`` to set the Java Truststore password used for https in internal communications between nodes.
+
+Hive Connector Changes
+________________
+* Add session property ``temporary_table_create_empty_bucket_files`` and configuration property ``hive.create-empty-bucket-files-for-temporary-table``, which, when set to false, disables the creation of zero-row files for temporary table empty buckets, to improve performance.
```
Comment (mayankgarg1990) [inline presto-docs/src/main/sphinx/release/release-0.248.rst:28]:
```
``false``
```

**Review by mayankgarg1990 — CHANGES_REQUESTED** [review]

File: presto-docs/src/main/sphinx/release/release-0.248.rst (hunk 8 above)
Comment (rschlussel) [reply presto-docs/src/main/sphinx/release/release-0.248.rst:21]: There's no documentation for this yet, so we can't add the function link.

File: presto-docs/src/main/sphinx/release/release-0.248.rst (hunk 5 above)
Comment (prithvip) [reply presto-docs/src/main/sphinx/release/release-0.248.rst:7]: done

File: presto-docs/src/main/sphinx/release/release-0.248.rst (hunk 6 above)
Comment (prithvip) [reply presto-docs/src/main/sphinx/release/release-0.248.rst:9]: done

File: presto-docs/src/main/sphinx/release/release-0.248.rst (hunk 7 above)
Comment (prithvip) [reply presto-docs/src/main/sphinx/release/release-0.248.rst:16]: The release notes guideline state that all optimizations must start with "Improve". https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines#order-of-changes
I've changed the phrasing to say "Improve query performance by reducing lock contention in output buffer memory tracking" for more clarity on the effect of the change

File: presto-docs/src/main/sphinx/release/release-0.248.rst (hunk 8 above)
Comment (prithvip) [reply presto-docs/src/main/sphinx/release/release-0.248.rst:21]: destructure_tdigest doesn't have any documentation, but I've surrounded TDigest and ROW with the double back quotes.

File: presto-docs/src/main/sphinx/release/release-0.248.rst (hunk 9 above)
Comment (prithvip) [reply presto-docs/src/main/sphinx/release/release-0.248.rst:28]: done

File: presto-docs/src/main/sphinx/release/release-0.248.rst (hunk 8 above)
Comment (mayankgarg1990) [reply presto-docs/src/main/sphinx/release/release-0.248.rst]: As per a conversation with @rschlussel - we should not include any function mention in the release notes if there is no documentation. lets remove the mention of this function from here and the highlights as well

Comment (prithvip) [reply presto-docs/src/main/sphinx/release/release-0.248.rst]: Ok, I've removed references to destructure_tdigest both in details and highlights section

**Review by mayankgarg1990 — APPROVED** [review]

**Review by rschlussel — APPROVED** [review]

**Pull request merged**